invenio/modules/workflows/models.py

```diff
@@ -682,16 +727,15 @@ def change_status(self, message):
 
     def get_current_task(self):
         """Return the current task from the workflow engine for this object."""
-        extra_data = self.get_extra_data()
         try:
-            return extra_data["_task_counter"]
+            return self.extra_data["_task_counter"]
         except KeyError:
             # Assume old version "task_counter"
-            return extra_data["task_counter"]
+            return self.extra_data["task_counter"]
```

Comment:
Sounds good in regards to task_counter.

Regarding Holding Pen, I think I would prefer that the Invenio DB model we'll use has this distinction at top-level as this is quite important for the Holding Pen/Invenio use case.